library/azure_rm_roledefinition.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu, (@yungezz)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_roledefinition
+version_added: "2.8"
+short_description: Manage Azure Role Definition.
+description:
+    - Create, update and delete instance of Azure Role Definition.
+
+options:
+    name:
+        description:
+            - Unique name of role definition.
+        required: True
+    permissions:
+        description:
+            - Set of role definition peremissions.
+            - See U(https://docs.microsoft.com/en-us/azure/app-service/app-service-web-overview) for more info.
+        suboptions:
+            actions:
+                description:
+                    - List of allowed actions.
+                type: list
+            not_actions:
+                description:
+                    - List of denied actions.
+                type: list
+            data_actions:
+                description:
+                    - List of allowed data actions.
+                type: list
+            not_data_actions:
+                description:
+                    - List of denied data actions.
+                type: list
+    assignable_scopes:
+        description: List of assignable scope of this definition.
+    scope:
+        description: The scope of the role definition.
+    description:
+        description:
+            - The role definition description.
+    state:
+      description:
+        - Assert the state of the role definition.
+        - Use 'present' to create or update a role definition and 'absent' to delete it.
+      default: present
+      choices:
+        - absent
+        - present
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Yunge Zhu(@yungezz)"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a role definition
+      azure_rm_roledefinition:
+        name: myTestRole
+        scope: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myresourceGroup
+        permissions:
+            actions:
+                - "Microsoft.Compute/read"
+                - "Microsoft.Storage/read"
+            data_actions:
+                - "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write"
+        assignable_scopes:
+            - "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+
+'''
+
+RETURN = '''
+id:
+    description: Id of current role definition.
+    returned: always
+    type: str
+    sample: {
+        "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/providers/Microsoft.Authorization/roleDefinitions/roleDefinitionId"
```

Comment:
why in {} --- it should be just:
sample: "/subscription/….." ?

Reply:
fixed